### PR TITLE
GitHub loader

### LINKED ISSUE #861: `GithubPackageMeshLoader` for multiple mesh packages
**Describe the bug**
RFL model loading issue: `GithubPackageMeshLoader` requires a package name, but for repos that are multi-robot and have multiple package sources, it does not work.

**To Reproduce**
1. Context:  Rhino GH
2. Sample script:
```
import compas
from compas.robots import GithubPackageMeshLoader
from compas.robots import RobotModel

# Set high precision to import meshes defined in meters
compas.PRECISION = '12f'

# Select Github repository, package and branch where the model is stored
# abb arm model
abb_repository = '.../...'   # non-public repo
abb_package = 'abb_irb4600_40_255'
abb_branch = 'master'
abb_github = GithubPackageMeshLoader(abb_repository, abb_package, abb_branch)
abb_urdf = abb_github.load_urdf('abb_irb4600_40_255.urdf')


# rfl model
rfl_repository = '.../...'   # non-public repo
rfl_package = 'rfl_description'
rfl_branch = 'master'
rfl_github = GithubPackageMeshLoader(rfl_repository, rfl_package, rfl_branch)
rfl_urdf = rfl_github.load_urdf('rfl.urdf')


# Create robot model from URDF
model = RobotModel.from_urdf_file(rfl_urdf)

# Also load geometry
model.load_geometry(rfl_urdf, abb_urdf)
```

```
Runtime error (Exception): HTTPError

Traceback:
  line 407, in _call_chain, "C:\Program Files\Rhino 6\Plug-ins\IronPython\Lib\urllib2.py"
  line 473, in error, "C:\Program Files\Rhino 6\Plug-ins\IronPython\Lib\urllib2.py"
  line 547, in http_response, "C:\Program Files\Rhino 6\Plug-ins\IronPython\Lib\urllib2.py"
  line 556, in http_error_default, "C:\Program Files\Rhino 6\Plug-ins\IronPython\Lib\urllib2.py"
  line 435, in open, "C:\Program Files\Rhino 6\Plug-ins\IronPython\Lib\urllib2.py"
  line 154, in urlopen, "C:\Program Files\Rhino 6\Plug-ins\IronPython\Lib\urllib2.py"
  line 71, in load_urdf, "C:\Users\pacherm\AppData\Roaming\McNeel\Rhinoceros\6.0\scripts\compas\robots\resources\github.py"
  line 15, in script
```

**Expected behavior**
Load a robot model from GitHub repository.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Python version: IronPython 2.7
 - Python package manager: conda

### PR DESCRIPTION
**Pull request by beverlylytle**
Closes #861  I used the following script to expose the issue and fix it:
```
import compas
from compas.robots import GithubPackageMeshLoader
from compas.robots import RobotModel
from compas_rhino.artists import RobotModelArtist

# Set high precision to import meshes defined in meters
compas.PRECISION = '12f'

# Select Github repository, package and branch where the model is stored
# abb arm model
abb_repository = 'gramaziokohler/abb_irb4600_40_255'
abb_package = 'abb_irb4600_40_255'
abb_branch = 'master'
abb_github = GithubPackageMeshLoader(abb_repository, abb_package, abb_branch, '.')
abb_urdf = abb_github.load_urdf('abb_irb4600_40_255.urdf')


# rfl model
rfl_repository = 'gramaziokohler/rfl_description'
rfl_package = 'rfl_description'
rfl_branch = 'master'
rfl_github = GithubPackageMeshLoader(rfl_repository, rfl_package, rfl_branch)
rfl_urdf = rfl_github.load_urdf('rfl.urdf')


# Create robot model from URDF
model = RobotModel.from_urdf_file(rfl_urdf)

# Also load geometry
model.load_geometry(rfl_github, abb_github)

artist = RobotModelArtist(model)
artist.draw()
```

### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [x] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas/robots/resources/github.py [hunk 1]
```diff
@@ -102,8 +110,4 @@ def load_mesh(self, url):
         _prefix, path = url.split(self.schema_prefix)
         url = self.build_url(path)
 
-        # TODO: As soon as compas.files adds support
-        # for file-like objects, we could skip
-        # storing a temp file for these urls
-        tempfile, _ = urlretrieve(url)
-        return _mesh_import(url, tempfile)
+        return _mesh_import(url, url)
```
Comment (beverlylytle) [inline src/compas/robots/resources/github.py:113]: yay!

File: src/compas/robots/resources/github.py [hunk 2]
```diff
@@ -52,11 +56,15 @@ def build_url(self, file):
...
-                                       self.support_package,
-                                       file)
+        url_components = [
+            GithubPackageMeshLoader.HOST,
+            self.repository,
+            self.branch,
+            file
+        ]
+        if self.relative_path != '.':
+            url_components.insert(3, self.relative_path)
```
Comment (gonzalocasas) [inline src/compas/robots/resources/github.py:66]: The `insert(3, [..]` feels fragile. How about doing something like:
```python
relative_path_component = self.relative_path if self.relative_path != '.' else None
url_components = [
   GithubPackageMeshLoader.HOST,
   self.repository,
   self.branch,
   relative_path_component,
   file
]

return '/'.join(filter(None, url_components))
```

File: src/compas/robots/resources/github.py (hunk 1 above)
Comment (gonzalocasas) [reply src/compas/robots/resources/github.py:113]: 💪

File: CHANGELOG.md [hunk 3]
```diff
@@ -13,9 +13,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 * `compas.robots.Axis` is now normalized upon initialization.
 * Fixed a bug in `compas.numerical.dr_numpy` when using numpy array as inputs.
+* `compas.robots.GithubPackageMeshLoader` allows for varying repository file structures.
```
Comment (gonzalocasas) [inline CHANGELOG.md]: Absolute nitpicking alert.

I'd write changelog lines always starting with the verb in past tense (Fixed, Added, Changed, Removed, Allowed, etc)

`end-of-nitpicking`

**Review by gonzalocasas — APPROVED** [review]
LGTM!

File: src/compas/robots/resources/github.py (hunk 2 above)
Comment (beverlylytle) [reply src/compas/robots/resources/github.py:66]: oh, somehow I completely overlooked this comment before merging.  sorry.  i'll revisit this later.

**Pull request merged**